## 50_session_enable_custom.diff by Peter S Galbraith <psg@debian.org>

--- a/elisp/emacs-goodies-el/session.el
+++ b/elisp/emacs-goodies-el/session.el
@@ -244,7 +244,12 @@
 		      (const :tag "Load/Save Session" session)
 		      (const :tag "Store/Use Places" places)
 		      (const :tag "Setup Key/Mouse Bindings" keys)
-		      (const :tag "Setup Menus" menus))))
+		      (const :tag "Setup Menus" menus)))
+  :require 'session
+  :set (lambda (symbol value)
+         (set-default symbol value)
+         (when value
+           (add-hook 'after-init-hook 'session-initialize))))
 
 
 ;;;===========================================================================
